packager/media/codecs/avc_decoder_configuration_record.cc

```diff
@@ -82,6 +82,31 @@ bool AVCDecoderConfigurationRecord::ParseInternal() {
     AddNalu(nalu);
   }
 
+  if (profile_indication_ == 100 || profile_indication_ == 110 || 
+      profile_indication_ == 122 || profile_indication_ == 144) {
+
+    uint8_t sps_ext_count;
+    if (!reader.Read1(&chroma_format_) || !reader.Read1(&bit_depth_luma_minus8_) ||
+        !reader.Read1(&bit_depth_chroma_minus8_) || !reader.Read1(&sps_ext_count)) {
+       LOG(WARNING) << "Insufficient bits in bitstream for given AVC profile";
+       return true;
+    }
+    chroma_format_ &= 0x3;
+    bit_depth_luma_minus8_ &= 0x7;
+    bit_depth_chroma_minus8_ &= 0x7;
+  
+    for (uint8_t i = 0; i < sps_ext_count; i++) {
+      uint16_t size = 0;
+      RCHECK(reader.Read2(&size));
+      const uint8_t* nalu_data = reader.data() + reader.pos();
+      RCHECK(reader.SkipBytes(size));
+
+      Nalu nalu;
+      RCHECK(nalu.Initialize(Nalu::kH264, nalu_data, size));
+      RCHECK(nalu.type() == Nalu::H264_SPSExtension);
+      AddNalu(nalu);
+    } 
+  } 	
```

Comment:
Can you update avc_decoder_configuration_record_unittest.cc to test the parsing done here? You can get the decoder configuration from a real content or you can make it up yourself?